src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs

```diff
@@ -105,30 +110,50 @@ public string ServicePrincipalSecret()
         ///GENMHASH:6B9110585F8C0802DD175E2A4CC42085:0AFA47BFFB09777DA247B843C7A8F6D9
         public byte[] AdminKubeConfigContent()
         {
-            if (this.Inner.AccessProfiles == null
-                || this.Inner.AccessProfiles.ClusterAdmin == null
-                || this.Inner.AccessProfiles.ClusterAdmin.KubeConfig == null)
+            if (this.adminKubeConfigContent == null)
+            {
+                this.adminKubeConfigContent = Extensions.Synchronize(() => this.GetAdminKubeConfigContent());
+            }
+
+            return this.adminKubeConfigContent;
+        }
+
+        ///GENMHASH:94A9F49AECE7874D497A7F5E8A5AB25C:FB91DAFFC2B4B802AB4DE52634A023D9
+        public byte[] UserKubeConfigContent()
+        {
+            if (this.userKubeConfigContent == null)
+            {
+                this.userKubeConfigContent = Extensions.Synchronize(() => this.GetUserKubeConfigContent());
+            }
+
+            return this.userKubeConfigContent;
+        }
+
+        private async Task<byte[]> GetAdminKubeConfigContent()
+        {
+            var profileInner = await this.Manager.Inner.ManagedClusters
+                .GetAccessProfilesAsync(this.ResourceGroupName, this.Name, KubernetesClusterAccessProfileRole.ADMIN.ToString());
+            if (profileInner == null)
             {
                 return new byte[0];
             }
             else
             {
-                return System.Convert.FromBase64String(this.Inner.AccessProfiles.ClusterAdmin.KubeConfig);
+                return System.Convert.FromBase64String(profileInner.KubeConfig);
             }
         }
 
-        ///GENMHASH:94A9F49AECE7874D497A7F5E8A5AB25C:FB91DAFFC2B4B802AB4DE52634A023D9
-        public byte[] UserKubeConfigContent()
+        private async Task<byte[]> GetUserKubeConfigContent()
```

Comment:
>GetUserKubeConfigContent [](start = 35, length = 24)

This one as well - please add Async suffix to the method name.

Reply:
Fixed.